BundleClient/app/src/main/java/net/discdd/bundleclient/screens/HomeScreen.kt

```diff
@@ -0,0 +1,148 @@
+        listOf(
+            TabItem(
+                title = context.getString(R.string.home_tab),
+                screen = { WifiDirectScreen(serviceReadyFuture = WifiServiceManager.serviceReady) }
+                ),
+            TabItem(
+                title = context.getString(R.string.server_tab),
+                screen = { ServerScreen() }
+            ),
+            TabItem(
+                title = context.getString(R.string.logs_tab),
+                screen = { LogScreen() }
+            ),
+            TabItem(
+                title = context.getString(R.string.bm_tab),
+                screen = { ManagerScreen() }
+            ),
+            TabItem(
+                title = context.getString(R.string.permissions_tab),
+                screen = { PermissionScreen(permissionsViewModel, activityResultLauncher) }
+            )
+        )
+    }
+    val usbTab = listOf(TabItem(
+        title = context.getString(R.string.usb_tab),
+        screen = { UsbScreen() }
+    ))
+
+    var tabItems by remember {
+        mutableStateOf(standardTabs)
+    }
+    LaunchedEffect(showUsbScreen) {
+        tabItems = if (showUsbScreen) {
+             usbTab + standardTabs
+        } else {
+            standardTabs
+        }
+    }
+
+    Surface(
+        modifier = Modifier.fillMaxSize(),
+        color = MaterialTheme.colorScheme.background
+    ) {
+        val pagerState = rememberPagerState() { tabItems.size }
+        val selectedTabIndex by remember {
+            derivedStateOf { pagerState.currentPage }
+        }
+        Column(
+            modifier = Modifier.fillMaxSize()
+        ) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp, vertical = 8.dp),
+            ) {
+                Text(
+                    text = context.getString(R.string.app_name),
+                    style = MaterialTheme.typography.titleLarge
+                )
+            }
+
+            ScrollableTabRow(
+                selectedTabIndex = selectedTabIndex,
+                edgePadding = 0.dp
+            ) {
+                tabItems.forEachIndexed { index, item ->
+                    Tab(
+                        selected = index == selectedTabIndex,
+                        onClick = {
+                            scope.launch {
+                                pagerState.animateScrollToPage(index)
+                            }
+                        },
+                        text = {
+                            Text(text = item.title)
+                        }
+                    )
+                }
+            }
+
+            HorizontalPager(
+                state = pagerState,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .weight(1f)
+            ) { index ->
+                tabItems[index].screen()
+            }
+        }
+    }
+}
+
+@Preview(showBackground = true)
```

Comment:
we should probably remove this function, correct? or pass in some bogus values so that it works.